src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -47,40 +46,32 @@ internal class InstallToolCommand : CommandBase
             {
                 throw new ArgumentNullException(nameof(appliedCommand));
             }
-
             _packageId = appliedCommand.Arguments.Single();
             _packageVersion = appliedCommand.ValueOrDefault<string>("version");
             _configFilePath = appliedCommand.ValueOrDefault<string>("configfile");
             _framework = appliedCommand.ValueOrDefault<string>("framework");
             _source = appliedCommand.ValueOrDefault<string>("source");
             _global = appliedCommand.ValueOrDefault<bool>("global");
             _verbosity = appliedCommand.SingleArgumentOrDefault("verbosity");
+            _toolPath = appliedCommand.SingleArgumentOrDefault("tool-path");
 
             var cliFolderPathCalculator = new CliFolderPathCalculator();
 
-            _toolPackageStore = toolPackageStore
-                ?? new ToolPackageStore(new DirectoryPath(cliFolderPathCalculator.ToolsPackagePath));
-
-            _toolPackageInstaller = toolPackageInstaller
-                ?? new ToolPackageInstaller(
-                    _toolPackageStore,
-                    new ProjectRestorer(_reporter));
+            _toolPackageFactory = toolPackageFactory ?? new ToolPackageFactory();
 
             _environmentPathInstruction = environmentPathInstruction
                 ?? EnvironmentPathFactory.CreateEnvironmentPathInstruction();
-
-            _shellShimRepository = shellShimRepository
-                ?? new ShellShimRepository(new DirectoryPath(cliFolderPathCalculator.ToolsShimPath));
+            _shellShimRepositoryFactory = shellShimRepositoryFactory ?? new ShellShimRepositoryFactory();
 
             _reporter = (reporter ?? Reporter.Output);
             _errorReporter = (reporter ?? Reporter.Error);
         }
 
         public override int Execute()
         {
-            if (!_global)
+            if (_toolPath != null && _global)
             {
-                throw new GracefulException(LocalizableStrings.InstallToolCommandOnlySupportGlobal);
+                throw new GracefulException("Cannot have global and tool-path as opinion at the same time."); // TODO wul no checkin loc
             }
```

Comment:
Until we come up with a reasonable default for `--tool-path`, should we need a check for `!_global && string.IsNullOrEmpty(_toolPath)` and error that the `--tool-path` option is required?

Reply:
good point

Reply:
fixed